src/Sets/Zonotope.jl

```diff
@@ -371,19 +371,179 @@ nonlinear systems with uncertain parameters using conservative linearization.
 In Proc. of the 47th IEEE Conference on Decision and Control.*
 """
 function split(Z::Zonotope, j::Int)
-    @assert 1 <= j <= ngens(Z) "cannot split a zonotope with $(ngens(Z)) generators along index $j"
     c, G = Z.center, Z.generators
-    Gj = G[:, j]
-    Gj_half = Gj / 2
-
-    c₁ = c - Gj_half
-    c₂ = c + Gj_half
-
-    G₁ = copy(G)
-    G₁[:, j] = Gj_half
-    G₂ = copy(G₁)
 
+    c₁ = similar(c)
+    G₁ = similar(G)
     Z₁ = Zonotope(c₁, G₁)
+
+    c₂ = similar(c)
+    G₂ = similar(G)
     Z₂ = Zonotope(c₂, G₂)
+
+    split!(Z₁, Z₂, Z, j)
+end
+
+function split!(Z₁::Zonotope, Z₂::Zonotope, Z::Zonotope, j::Int)
+    c, G = Z.center, Z.generators
+    n, p = size(G)
+    @assert 1 <= j <= p "cannot split a zonotope with $p generators along index $j"
+
+    c₁, G₁ = Z₁.center, Z₁.generators
+    c₂, G₂ = Z₂.center, Z₂.generators
+    copyto!(G₁, G)
+    copyto!(G₂, G)
+
+    @inbounds for i in 1:n
+        α = G[i, j] / 2
+        c₁[i] = c[i] - α
+        c₂[i] = c[i] + α
+        G₁[i, j] = α
+        G₂[i, j] = α
+    end
+    return Z₁, Z₂
+end
+
+function load_static_arrays()
+return quote
+
+using .StaticArrays: SMatrix, SVector, MMatrix, MVector
+
+"""
+    split(Z::Zonotope{N, SVector{n, N}, <:SMatrix{n, p, N}}, j::Int) where {N, n, p}
+
+Return two zonotopes obtained by splitting the given zonotope.
+
+### Input
+
+- `Z` -- zonotope
+- `j` -- index of the generator to be split
+
+### Output
+
+The zonotopes obtained by splitting `Z` into two zonotopes such that
+their union is `Z` and their intersection is possibly non-empty.
+
+### Algorithm
+
+This function implements [Prop. 3, 1], that we state next. The zonotope
+``Z = ⟨c, g^{(1, …, p)}⟩`` is split into:
+
+```math
+Z₁ = ⟨c - \\frac{1}{2}g^{(j)}, (g^{(1, …,j-1)}, \\frac{1}{2}g^{(j)}, g^{(j+1, …, p)})⟩ \\\\
+Z₂ = ⟨c + \\frac{1}{2}g^{(j)}, (g^{(1, …,j-1)}, \\frac{1}{2}g^{(j)}, g^{(j+1, …, p)})⟩,
+```
+such that ``Z₁ ∪ Z₂ = Z`` and ``Z₁ ∩ Z₂ = Z^*``, where
+
+```math
+Z^* = ⟨c, (g^{(1,…,j-1)}, g^{(j+1,…, p)})⟩.
+```
+
+[1] *Althoff, M., Stursberg, O., & Buss, M. (2008). Reachability analysis of
+nonlinear systems with uncertain parameters using conservative linearization.
+In Proc. of the 47th IEEE Conference on Decision and Control.*
+"""
+function split(Z::Zonotope{N, SV, SM}, j::Int) where {N, n, p, SV<:SVector{n, N}, SM<:SMatrix{n, p, N}}
```

Comment:
I tried this out but for me it does not choose this method.
```julia
julia> Z
Zonotope{Float64,SArray{Tuple{2},Float64,1,2},SArray{Tuple{2,2},Float64,2,4}}([0.7504943082957674, 0.36528999613841906], [-0.5419307531431913 1.1459414507001595; -0.08228095036507727 -0.4595899030827082])

julia> @which split(Z, 1)
split(Z::Zonotope, j::Int64) in LazySets at .julia/dev/LazySets/src/Sets/Zonotope.jl:374
```
Are you sure about the signature? In my example, I think it should have the signatures `SV<:SVector{Tuple{n}, N, 1, n}` and `SM<:SMatrix{Tuple{n, p}, N, n, p}` (not 100% sure this is valid syntax).

Reply:
StaticArrays has some convenience type aliases, eg. 

```julia
julia> MVector{2, Float64}
MArray{Tuple{2},Float64,1,2}

julia> MMatrix{2, 3, Float64}
MArray{Tuple{2,3},Float64,2,L} where L
```

Reply:
> I tried this out but for me it does not choose this method.

i think this was because the uuid was wrong under `__init__`

Reply:
ill mark this as resolved, please reopen if it's not the case.